erc721-tests/contracts/Erc-721.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "hardhat/console.sol";
+
+contract Erc721 is Ownable, ERC721("SolicyNFT", "HNFT") {
+    uint256 tokenId = 0;
+    string public _name;
+    string public _symbol;
+    string public __baseURI;
+    string public uriSuffix = ".json";
+    mapping(address => TokenMetaData[]) public ownershipRecord;
+    mapping(uint256 => bool) internal lockRecord;
+
+    struct TokenMetaData {
```

Comment:
Could store something like lockedUntil variable, indicating timestamp, instead of lockedTimeStamp and daysToLock

Reply:
No we need that variables in separate